org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/opentelemetry/OpenTelemetryExtensionInstrumenterFactory.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.client.opentelemetry;
+
+import static io.opentelemetry.instrumentation.api.internal.AttributesExtractorUtil.internalSet;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import org.eclipse.scout.rt.platform.ApplicationScoped;
+import org.eclipse.scout.rt.platform.config.CONFIG;
+import org.eclipse.scout.rt.platform.opentelemetry.OpenTelemetryProperties.OpenTelemetryTracingEnabledProperty;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+
+@ApplicationScoped
+public class OpenTelemetryExtensionInstrumenterFactory {
+
+  public <OWNER> Instrumenter<OpenTelemetryExtensionRequest<OWNER>, Void> createInstrumenter(String instrumentationName, String prefix,
+      Function<OpenTelemetryExtensionRequest<OWNER>, String> getTextFunction) {
+    SpanNameExtractor<OpenTelemetryExtensionRequest<OWNER>> spanNameExtractor = request -> request.getOwner().getClass().getSimpleName() + "." + request.getEventName();
+    AttributesExtractor<OpenTelemetryExtensionRequest<OWNER>, Void> attributesExtractor = new AttributesExtractor<>() {
+      private final AttributeKey<String> m_requestClass = io.opentelemetry.api.common.AttributeKey.stringKey(prefix + ".class");
+      private final AttributeKey<String> m_requestName = AttributeKey.stringKey(prefix + ".text");
+      private final AttributeKey<String> m_eventName = AttributeKey.stringKey(prefix + ".event.name");
+
+      @Override
+      public void onStart(AttributesBuilder attributes, Context parentContext, OpenTelemetryExtensionRequest<OWNER> request) {
+        internalSet(attributes, m_requestClass, request.getOwner().getClass().getName());
```

Comment:
`internalSet` can be removed. `AttributesBuilder.put` provides the same function and is public API.
```suggestion
        attributes.put(m_requestClass, request.getOwner().getClass().getName());
```